docs/3.0/develop/transactions.mdx

```diff
@@ -119,3 +125,61 @@ def confirmation(transaction):
     print("committing a record now using the task's cache key!")
 ```
 </Tip>
+
+### Access data within transactions
+
+Key-value pairs can be set within a transaction and accessed elsewhere within the transaction, including within the `on_rollback` hook.
+
+The code below shows how to set a key-value pair within a transaction and access it within the `on_rollback` hook and the `quality_test` task: 
+
+```python
+import os
+from time import sleep
+
+from prefect import task, flow
+from prefect.transactions import get_transaction, transaction
+
+
+@task
+def write_file(contents: str):
+    "Writes to a file."
+    with open("side-effect.txt", "w") as f:
+        f.write(contents)
+
+
+@write_file.on_rollback
+def del_file(transaction):
+    "Deletes file."
+    os.unlink("side-effect.txt")
+    txn1 = get_transaction()      # get the transaction object
+    print(txn1.get("contents"))   # prints "hello world"
+
+
+@task
+def quality_test():
+    "Checks contents of file."
+    with open("side-effect.txt", "r") as f:
+        data = f.readlines()
+
+    if len(data) < 2:
+        txn2 = get_transaction()      # get the transaction object
+        raise ValueError(f"Not enough data! {(txn2.get("contents"))}") 
+        # prints "hello world" as part of the error message
+
+
+@flow
+def pipeline(contents: str):
+    with transaction() as txn:
+        write_file(contents)
+        sleep(2)
+        txn.set("contents", contents)  # set the contents key to the value of the contents variable
```

Comment:
i think this example is using the transaction key as a parameter; really the only reason to use `get` / `set` currently is for on-rollback and on-commit hooks, I don't recommend using transactions for data sharing across tasks

Reply:
Gotcha. Okay, will update. Thanks.